src/main/kotlin/minesweeper/model/Board.kt

```diff
@@ -1,18 +1,43 @@
 package minesweeper.model
 
-/**
- * @author 이상준
- */
 class Board(
-    private val height: Int,
-    private val width: Int,
+    height: Int = 0,
+    width: Int = 0,
     private val mineCount: Int,
 ) {
+    var height: Int = height
+        private set
+    var width: Int = width
+        private set
     private var cells = Cells()
 
-    fun createCells(): List<Cell> {
+    init {
+        require(height > 0 || width > 0 || mineCount > 0) { "높이, 너비, 지뢰 개수는 0보다 커야합니다." }
+    }
+
+    fun createCells(minesStrategy: MinesStrategy): Board {
         this.cells = cells.addAll(List(height * width) { index -> Cell(index / width, index % width) })
-        cells.addMines(mineCount)
-        return cells.cells
+        this.cells = minesStrategy.addMines(this.cells, mineCount)
+        return this
+    }
+
+    fun size(): Int {
+        return height * width
+    }
+
+    fun addMineAroundCounts() {
+        cells.addMineAroundCounts()
+    }
+
+    fun getCells(): Cells {
+        return cells
+    }
+
+    fun getRowCells(row: Int): Cells {
+        return Cells(cells.cellList.filter { it.row == row })
```

Comment:
cell을 필터하고 새로운 셀을 만드는 책임은 변환과 생성이라는 두 가지 책임이 섞여있는데 dot(.)을 보면 객체지향적이지 않은 설계로 보일 수 있습니다. 

일급 컬렉션을 활용해보면 어떨까요? 사실 SRP라는 개념이 모호해서 힘들지만, 이 경우엔 일급 컬렉션에 좀 더 책임을 위임해도 괜찮을 것 같네요.

```suggestion
        return cells.filter { it.row == row}.let { Cells(it) }
```

Reply:
네네 그부분이 더좋겠네요~